extensions/amp-analytics/0.1/scroll-manager.js

```diff
@@ -132,4 +181,30 @@ export class ScrollManager {
       this.onScroll_.bind(this)
     );
   }
+
+  /**
+   * Gets the cached layout rectangle of the root element
+   * @return {!Promise<!../../../src/layout-rect.LayoutRectDef>}
+   */
+  getInitRootElementRect_() {
+    if (this.initialRootLayoutRect_) {
+      return Promise.resolve(this.initialRootLayoutRect_);
+    }
+
+    return this.measureRootElement_();
+  }
+
+  /**
+   * Gets the layout rectangle of the root element
+   * @return {!Promise<!../../../src/layout-rect.LayoutRectDef>}
+   */
+  measureRootElement_() {
+    return this.mutator_.measureElement(() => {
+      const layoutRect = this.viewport_.getLayoutRect(this.root_);
+      if (!this.initialRootLayoutRect_) {
+        this.initialRootLayoutRect_ = layoutRect;
```

Comment:
to continue the discussion about the `initialScrollHeight` here https://github.com/ampproject/amphtml/pull/26451#discussion_r373707153. I can't reply to that thread

1. to confirm for `<amp-next-page>`, one only want to get scroll percentage of the next page right. the percentage will also start from 0 to 100. In that case initiating the scrollHeight too late can still be an issue for the next page if it contains re-sizable content. 
2. `DOMContentLoaded` can fire very late especially when streaming long documents.

I don't see good solution to these concerns. Best we can do is to remind developers of the caveat. Could you please add a paragraph to the `ignoreResize` doc. Thanks

Reply:
Added a note :) !